### PR TITLE
Update to Gradle 8.12

### PR DESCRIPTION
**Pull request by wmdietl**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -109,6 +109,12 @@ if (jspecify != null) {
     assemble.dependsOn(jspecify.task(':assemble'))
 }
 
+// Enable exec/javaexec
```
Comment (wmdietl) [inline build.gradle:112]: I'm happy for suggestions for how to do this in a nicer way... The [upgrade instructions](https://docs.gradle.org/8.12/userguide/upgrading_version_8.html#deprecated_project_exec) didn't really help me a lot.

Comment (cpovirk) [reply build.gradle:112]: Sadly but unsurprisingly, I'm clueless. Now that I know to look for `ExecOperations`, I see https://github.com/gradle/gradle/issues/30822, which seems encouraging, I think?

**Review by cpovirk — APPROVED** [review]
Thanks for taking care of this.

File: build.gradle (hunk 1 above)
Comment (wmdietl) [reply build.gradle:112]: Thanks, here I was doing the right (oddly verbose) thing but in `settings.gradle` I wasn't actually executing the task. Now it both compiles and seems to do the right thing.

**Pull request merged**